src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -374,11 +397,43 @@ async def _update_repo_using_branch_head(
     return f"{repo.repo_id}:{repo.branch}:{sha}"
 
 
-async def _check_repositories(repos: list[GitRepo]) -> dict:
+async def _check_repositories(
+    repos: list[GitRepo], syncedViaTags: bool = False
+) -> dict:
     changes = {}
     for repo in repos:
         log.debug("fetching repo: %s...", repo.repo_url)
         await _git_fetch(repo.directory)
+    latestTags = [
+        {
+            repo.repo_id: await _git_get_latest_matching_tag_capture_groups(
+                repo.directory, repo.tags
+            )
+        }
+        for repo in repos
+    ]
+    uniqueLatestTags = list(
+        {
+            list(i.values())[0][0] if list(i.values())[0] else None
+            for i in latestTags
+            if i.values()
+        }
+    )
+    if syncedViaTags:
+        if len(uniqueLatestTags) > 1:
+            log.info("Repos did not match in their latest tag's first capture group!")
+            log.info(
+                "Latest (matching) tags per repo, displaying first regex capture group:"
+            )
+            for repo in latestTags:
+                log.info("%s: %s", list(repo.keys())[0], list(repo.values())[0][0])
+            log.info("Will only update those repos that have no tag-regex specified!")
+        elif len(uniqueLatestTags) == 1:
+            log.info("All synced repos have the same latest tag! Deploying....")
+    for repo in repos:
+        if syncedViaTags and len(uniqueLatestTags) > 1:
```

Comment:
MINOR: here ``syncedViaTags`` will always be True (see line 422)

Reply:
why not in one line?
```python
if syncedViaTags and len(uniqueLatestTags) > 1 and repo.tags:
    continue
```

Reply:
> MINOR: here `syncedViaTags` will always be True (see line 422)

this I dont understand, if len(uniqueLatestTags) == 0, so no tags are watched,  `syncedViaTags` can still be true, right?

Reply:
my bad, i thought I saw before the for loop within the if statment
![image](https://user-images.githubusercontent.com/32402063/223373659-4c724741-a3e2-49b1-99cf-0585eb05040a.png)